bot.go

```diff
@@ -319,13 +313,66 @@ func (b *Bot) ConnectAndRun() error {
 	return b.Run(c)
 }
 
-// Run starts the bot and loops until it dies. It accepts a
-// ReadWriter. If you wish to use the connection feature from the
-// config, use ConnectAndRun.
-func (b *Bot) Run(c io.ReadWriter) error {
-	var err error
+func (b *Bot) EnsurePlugin(name string) error {
+	loaded, ok := b.loadedPlugins[name]
+	if !ok {
+		return fmt.Errorf("Plugin %q not loaded", name)
+	}
+
+	// If it's already loaded, return nil
+	if loaded {
+		return nil
+	}
+
+	return b.loadPlugin(name)
+}
 
-	b.injector, err = b.registry.Load(b.config.Plugins, nil)
+func (b *Bot) loadPlugin(name string) error {
+	tmpLoadingContext := append(b.loadingContext, name)
+
+	if com.IsSliceContainsStr(b.loadingContext, name) {
+		return fmt.Errorf(
+			"Plugin load loop: %s",
+			strings.Join(tmpLoadingContext, ", "))
+	}
+
+	// Push the current plugin onto the stack
+	b.loadingContext = tmpLoadingContext
+
+	err := plugins[name](b)
```

Comment:
It's pretty unclear that this is where we load recursively--can you add a comment here to make this easier to follow?

Reply:
Yep, good point!